### PR TITLE
WIP - Storybook UI Library

### PR DESCRIPTION
**Pull request by codemist**
# References: 
Jira: [MNTOR-1621]

# Description

Add Storybook (https://storybook.js.org/) to project to view components/design system elements. This includes a Button element, Status Pill and the Exposure Card to test.

# Screenshot (if applicable)

image

# How to test

Run `npm run storybook` to preview.

# Checklist (Definition of Done)
- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/app/(nextjs_migration)/components/client/button.module.scss [hunk 1]
```diff
@@ -0,0 +1,30 @@
+@import "../../../tokens";
+
+.button {
+  font: $text-body-md;
+  font-weight: 500;
+  border: 0;
+  padding: $spacing-sm $spacing-md;
+    border-radius: $border-radius-sm;
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/button.module.scss:8]: nit: weird indenting.

File: src/app/(nextjs_migration)/components/client/Button.stories.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import type { Meta, StoryObj } from "@storybook/react";
+
+import { Button } from "./Button";
+
+// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
+const meta: Meta<typeof Button> = {
+  title: "Example/Button",
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/Button.stories.ts:11]: Per https://storybook.js.org/docs/react/writing-stories/naming-components-and-hierarchy#naming-stories:

```
  /* 👇 The title prop is optional.
   * See https://storybook.js.org/docs/react/configure/overview#configure-story-loading
   * to learn how to generate automatic titles
   */
```

File: src/app/(nextjs_migration)/components/client/button.module.scss [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  &.primary {
+    color: white;
+    background-color: $color-blue-50;
+  }
+
+  &.secondary {
+    color: $color-blue-50;
+      border: solid 2px $color-blue-50;
+      background-color: transparent;
+    box-shadow: rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset;
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/button.module.scss:22]: Same here (and in `&.large` below) re: weird indentation.
Not sure we're using prettier yet or if we need to check to see if it's watching .scss files.

File: src/app/package.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC"
+}
```
Comment (pdehaan) [inline src/app/package.json:12]: This feels a bit suspicious. Did we mean to commit a nested src/app/package.json (and package-lock.json) file?

File: src/app/storybook.log [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+npm ERR! code ERR_SOCKET_TIMEOUT
+npm ERR! network Socket timeout
+npm ERR! network This is a problem related to network connectivity.
+npm ERR! network In most cases you are behind a proxy or have bad network settings.
+npm ERR! network 
+npm ERR! network If you are behind a proxy, please make sure that the
+npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
+
+npm ERR! A complete log of this run can be found in:
+npm ERR!     /Users/kandres/.npm/_logs/2023-05-30T16_33_34_119Z-debug-0.log
```
Comment (pdehaan) [inline src/app/storybook.log:11]: We probably didn't mean to commit this file.

File: src/app/tokens.scss [hunk 6]
```diff
@@ -197,8 +197,8 @@ $text-title-2xs: 700 24px / 26px "Metropolis", sans-serif;
 $text-title-3xs: 700 18px / 20px "Metropolis", sans-serif;
```
Comment (pdehaan) [inline src/app/tokens.scss:197]: Re: no longer importing metropolis.scss on L1 above, do we still want to call it on L192-197 here? Or is it already being imported elsewhere?

```diff
-@import "./fonts/metropolis.scss";
+// @import "./fonts/metropolis.scss";
```

File: src/app/(nextjs_migration)/components/client/StatusPill.module.scss [hunk 7]
```diff
@@ -0,0 +1,27 @@
+@import "../../../tokens";
+
+.pill {
+  font: $text-body-sm;
+  font-weight: 600;
+  background: black;
+  display: inline-block;
+  color: white;
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/StatusPill.module.scss:8]:
```suggestion
  color: $color-white;
```

File: src/app/(nextjs_migration)/components/client/StatusPill.module.scss [hunk 8]
```diff
@@ -0,0 +1,27 @@
+@import "../../../tokens";
+
+.pill {
+  font: $text-body-sm;
+  font-weight: 600;
+  background: black;
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/StatusPill.module.scss:6]:
```suggestion
  background: $color-black;
```

File: src/app/(nextjs_migration)/components/client/button.module.scss [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  font-weight: 600;
+  border: 0;
+  padding: $spacing-md $spacing-lg;
+  border-radius: $border-radius-sm;
+  cursor: pointer;
+  display: inline-block;
+  line-height: 1;
+
+  &.primary {
+    color: white;
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/button.module.scss:14]:
```suggestion
    color: $color-white;
```

File: src/app/(nextjs_migration)/components/client/button.module.scss [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  &.secondary {
+    color: $color-blue-50;
+    border: solid 2px $color-blue-50;
+    background-color: transparent;
+    box-shadow: rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset;
+  }
+
+  &.large {
+    font: $text-body-lg;
+    color: white;
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/button.module.scss:27]:
```suggestion
    color: $color-white;
```

File: src/app/(nextjs_migration)/components/client/button.module.scss [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+  &.primary {
+    color: white;
+    background-color: $color-blue-50;
+  }
+
+  &.secondary {
+    color: $color-blue-50;
+    border: solid 2px $color-blue-50;
+    background-color: transparent;
+    box-shadow: rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset;
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/button.module.scss:22]:
```suggestion
    box-shadow: rgba($color-black, 0.15) 0 0 0 1px inset;
```

File: src/app/.storybook/preview.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
+/** @type { import('@storybook/react').Preview } */
```
Comment (pdehaan) [inline src/app/.storybook/preview.js:1]:
```suggestion
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

/** @type { import('@storybook/react').Preview } */
```

**Review by OrahNbur — COMMENTED** [review]
Hhsrsafety@gmail.com

File: src/app/fonts/inter.scss [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+https://web.dev/font-best-practices/
+https://web.dev/css-size-adjust/
+https://barrd.dev/article/create-a-variable-font-subset-for-smaller-file-size/
+https://fonts.googleapis.com/css2?family=Inter:wght@400;600&display=swap
+*/
+
+/* Arial font-size matching for minimal CLS */
+
+@font-face {
+  font-family: Inter-fallback;
```
Comment (pdehaan) [inline src/app/fonts/inter.scss:17]: Not relevant to much probably, but I was reading through the Next.js docs the other day and came across this:
https://nextjs.org/docs/pages/building-your-application/optimizing/fonts

```js
import { Inter } from 'next/font/google';
 
// If loading a variable font, you don't need to specify the font weight
const inter = Inter({ subsets: ['latin'] });
```

Not sure if that's worth filing a separate ticket/spike for to investigate in the future if we could potentially do font optimizing.

I also didn't see a Metropolis font on Google Fonts, so wasnt sure how to do potential font optimizing for locally hosted fonts.

File: .storybook/preview.ts [hunk 14]
```diff
@@ -0,0 +1,15 @@
+import type { Preview } from "@storybook/react";
```
[the remark this answers is not in the document]
Comment (maxxcrawford) [reply .storybook/preview.ts:5]: Fixed in 2c25ca3f92e8a2592695534c8d75b3e2771edbfa. (Added ESLint)

File: src/app/.storybook/preview.js (hunk 12 above)
Comment (maxxcrawford) [reply src/app/.storybook/preview.js:1]: Fixed in 2c25ca3f92e8a2592695534c8d75b3e2771edbfa. (Added ESLint)

File: .storybook/main.ts [hunk 15]
```diff
@@ -0,0 +1,17 @@
+import type { StorybookConfig } from "@storybook/nextjs";
```
[the remark this answers is not in the document]
Comment (maxxcrawford) [reply .storybook/main.ts:5]: Fixed in 2c25ca3f92e8a2592695534c8d75b3e2771edbfa. (Added ESLint)

File: src/app/.storybook/main.js [hunk 16]
```diff
@@ -0,0 +1,20 @@
+/** @type { import('@storybook/nextjs').StorybookConfig } */
```
[the remark this answers is not in the document]
Comment (maxxcrawford) [reply src/app/.storybook/main.js:1]: Fixed in 2c25ca3f92e8a2592695534c8d75b3e2771edbfa. (Added ESLint)

File: .storybook/preview.ts [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+  parameters: {
+    actions: { argTypesRegex: "^on[A-Z].*" },
+    controls: {
+      matchers: {
+        color: /(background|color)$/i,
+        date: /Date$/,
+      },
+    },
+    status: {
+      type: "beta", // 'beta' | 'stable' | 'deprecated' | 'releaseCandidate'
```
Comment (maxxcrawford) [inline .storybook/preview.ts:17]: Note: This currently applies to ALL components. We need to figure out how to set individual status levels.

**Comment by maxxcrawford** [comment]
I think we can review this as is. I added a "status" label add-on so that we can caveat which components are completed/etc. 

@pdehaan @codemist @Vinnl

File: src/app/(nextjs_migration)/components/client/ExposureCard.tsx [hunk 18]
```diff
@@ -0,0 +1,186 @@
...
+    statusPillContent,
+    statusPillType,
+  } = props;
+
+  const [detailsOpen, setDetailsOpen] = useState(false);
+
+  const DetailsFoundItem = (props: DetailsFoundProps) => {
+    let headline, description;
+    if (props.whichExposed === "family") {
+      headline = "Family members";
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/ExposureCard.tsx:53]: Not sure if we want to add `TODO` statements to refactor all this for l10n-ing.

File: src/app/(nextjs_migration)/components/client/ExposureCard.tsx [hunk 19]
```diff
@@ -0,0 +1,186 @@
...
+      }
+    }
+
+    return (
+      <>
+        <dt>
+          <span className={styles.exposureTypeIcon}>{props.icon}</span>
+          {headline}
+        </dt>
+        <dl>{description}</dl>
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/ExposureCard.tsx:94]: IIUC, i think `<dl>` is supposed to be a parent of `<dt>` and `<dd>` tags

File: src/app/(nextjs_migration)/components/client/ExposureCard.tsx [hunk 20]
```diff
@@ -0,0 +1,186 @@
...
+      </>
+    );
+  };
+  const elementCard = (
+    <div>
+      <div className={styles.exposureCard}>
+        <div className={styles.exposureHeader}>
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/ExposureCard.tsx:101]: nit: should this be `<div>` or `<header>`?

File: src/app/(nextjs_migration)/components/client/ExposureCard.tsx [hunk 21]
```diff
@@ -0,0 +1,186 @@
...
+        <dt>
+          <span className={styles.exposureTypeIcon}>{props.icon}</span>
+          {headline}
+        </dt>
+        <dl>{description}</dl>
+      </>
+    );
+  };
+  const elementCard = (
+    <div>
```
Comment (pdehaan) [inline src/app/(nextjs_migration)/components/client/ExposureCard.tsx:99]: nit: Do we need this wrapper `div`, or can we just use the child `<div className={styles.exposureCard}>` as the top level container?

File: src/app/.storybook/main.js [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+const config = {
+  stories: [
+    // "../components/client/stories/**/*.mdx",
+    "../components/client/stories/**/*.stories.@(js|jsx|ts|tsx)",
```
Comment (pdehaan) [inline src/app/.storybook/main.js:5]: Do we need all these globs, or can we reduce this to just .ts and/or .tsx?

**Comment by Vinnl** [comment]
[A wild preview deployment appeared!](https://storybook-ui--fx-monitor-storybook.netlify.app/)

File: src/app/fonts/inter.scss (hunk 13 above)
Comment (Vinnl) [reply src/app/fonts/inter.scss:17]: Yeah, and AFAIK we're already using that - Storybook is useful, but it shouldn't negatively impact what our actual website looks like. So if Storybook has issues loading the fonts, let's see if we can fix that in a way that doesn't require changes to the actual website code - and until we've found such a fix, we'll just have to accept that the fonts don't look quite right there yet.

**Review by Vinnl — COMMENTED** [review]
Really cool that this adds all the Storybook infra, this is going to be massively useful! Needs a bit of cleanup, but let's do that, then merge this in, and then we can work on separate components in followup pull requests as needed. To clean up to merge, I think we should:

- Revert the addition of unoptimised `Inter`
- Remove the second `.storybook` folder
- Move new components out of `/app/(nextjs_migration)`, e.g. to `/app/components/

Nice work!

**Review by Vinnl — APPROVED** [review]

**Pull request merged**